lntest/itest/lnd_onchain_test.go

```diff
@@ -379,3 +382,237 @@ func testAnchorReservedValue(net *lntest.NetworkHarness, t *harnessTest) {
+	// help sweep all coins.
+	net.SendCoins(t.t, secondChanSize, alice)
+
+	// Open the channel between the two nodes and wait for it to confirm
+	// fully.
+	aliceChanPoint1 := openChannelAndAssert(
+		t, net, alice, bob, lntest.OpenChannelParams{
+			Amt: firstChanSize,
+		},
+	)
+
+	// With the channel open, we'll actually immediately force close it. We
+	// don't care about network announcements here since there's no routing
+	// in this test.
+	_, _, err := net.CloseChannel(alice, aliceChanPoint1, true)
+	if err != nil {
+		t.Fatalf("unable to execute force channel closure: %v", err)
+	}
+
+	// Now that the channel has been force closed, it should show up in the
+	// PendingChannels RPC under the waiting close section.
+	pendingChansRequest := &lnrpc.PendingChannelsRequest{}
+	ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+	pendingChanResp, err := alice.PendingChannels(ctxt, pendingChansRequest)
+	if err != nil {
+		t.Fatalf("unable to query for pending channels: %v", err)
+	}
+	err = checkNumWaitingCloseChannels(pendingChanResp, 1)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	// Get the normal channel outpoint so we can track it in the set of
+	// channels that are waiting to be closed.
+	fundingTxID, err := lnrpc.GetChanPointFundingTxid(aliceChanPoint1)
+	if err != nil {
+		t.Fatalf("unable to get txid: %v", err)
+	}
+	chanPoint := wire.OutPoint{
+		Hash:  *fundingTxID,
+		Index: aliceChanPoint1.OutputIndex,
+	}
+	waitingClose, err := findWaitingCloseChannel(pendingChanResp, &chanPoint)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	// At this point, the channel is waiting close, and we have both the
+	// commitment transaction and anchor sweep in the mempool.
+	const expectedTxns = 2
+	sweepTxns, err := getNTxsFromMempool(
+		net.Miner.Client, expectedTxns, minerMempoolTimeout,
+	)
+	require.NoError(t.t, err, "no sweep txns in miner mempool")
+	aliceCloseTx := waitingClose.Commitments.LocalTxid
+	_, aliceAnchor := findCommitAndAnchor(t, net, sweepTxns, aliceCloseTx)
+
+	// We'll now mine _only_ the commitment force close transaction, as we
+	// want the anchor sweep to stay unconfirmed.
+	var emptyTime time.Time
+	forceCloseTxID, _ := chainhash.NewHashFromStr(aliceCloseTx)
+	commitTxn, err := net.Miner.Client.GetRawTransaction(
+		forceCloseTxID,
+	)
+	if err != nil {
+		t.Fatalf("unable to get transaction: %v", err)
+	}
+	_, err = net.Miner.GenerateAndSubmitBlock(
+		[]*btcutil.Tx{commitTxn}, -1, emptyTime,
+	)
+	if err != nil {
+		t.Fatalf("unable to generate block: %v", err)
+	}
+
+	// With the anchor output located, and the main commitment mined we'll
+	// instruct the wallet to send all coins in the wallet to a new address
+	// (to the miner), including unconfirmed change.
+	minerAddr, err := net.Miner.NewAddress()
+	if err != nil {
+		t.Fatalf("unable to create new miner addr: %v", err)
+	}
+	sweepReq := &lnrpc.SendCoinsRequest{
+		Addr:             minerAddr.String(),
+		SendAll:          true,
```

Comment:
Make a note in the comment that `SendAll` uses the sweeper under the hood?

Reply:
I don't think this matters mechanically for the test: even if this didn't use the sweeper, we'd still be spending that unconfirmed change it produced which is the scenario we're actually trying to reproduce here.